docs/threat_models/smart_contracts/multipool_autoswap/mpas_sequence.puml

```diff
@@ -0,0 +1,182 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+header Agoric Operating Systems
+footer Page %page% of %lastpage%
+title Multipool Auto Swap Sequence Diagram
+
+autonumber
+actor "MPAS Creator" as masc #lime
+actor "Pool Adder N" as pa #yellow
+actor "Pool User" as pu
+control "Zoe" as zoe
+database "MPAS Contract" as mpas
+
+== MPAS Creation ==
+masc -> mpas: Create instance
+return ""creatorFacet""
+== Pool Addition SML:BCK==
+autonumber 10
+pa -> mpas: ""addPool()""
+create database "Pool SML:BUCK" as pool
+mpas -> pool: Create pool for SML -> BUCK
+mpas --> pa: ""Issuer""
+== Pool Addition MOOL:BCK==
+autonumber 10
+pa -> mpas: ""addPool()""
+create database "Pool MOOL:BUCK" as pool2
+mpas -> pool2: Create pool for MOOL -> BUCK
+mpas --> pa: ""Issuer""
+
+
+== Duplicate Pool Addition MOOL:BCK==
+autonumber 30
+pa -> mpas: ""addPool()""
+mpas -->x pa: pool for MOOL:BCK already exists
+newpage
+autonumber 50
+== Swap Request ==
+group asking for the current price UNK:BUCK
+pu -> mpas: ""getOutputPrice(UNK:BUCK)""
+mpas -->x pu: ""brands were not recognized""
+end
+group asking for the current price MOOL:BUCK
+
+
+pu -> mpas: ""getOutputPrice(MOOL:BUCK)""
+mpas -> pool2: ""getPriceGivenRequiredOutput(MOOL:BUCK)""
+pool2 --> mpas: ""quote""
+mpas --> pu: ""quote""
+end
+
+group making a proposal
+pu -> pu: create proposal for SML:BUCK as ""proposal""
+
+end
+
+group requesting a SwapOut an invitation
+pu -> mpas: ""makeSwapOutInvitation()""
+return ""invitation""
+
+end
+
+group making an offer
+pu -> pu: withdraw funds from purse of SML into ""payment""
+pu -> zoe: ""offer(invitation, proposal, payment))""
+zoe -> zoe: validate offer
+create control "userSeat" as usA
+zoe -> usA: create ""userSeat""
+usA --> zoe: ""userSeat"" ref
+zoe -> pool: ""getPoolSeat()""
+create control "poolSeat" as psA
+pool -> psA: create ""poolSeat"" ref
+psA --> zoe: ""poolSeat"" ref
+zoe -> psA: add/remove funds
+zoe -> usA: add/remove funds
+zoe -> zoe: ""reallocate(seat, poolSeat)""
```

Comment:
I think showing this as interacting with Zoe is a mistake. There's a sense in which zoe itself creates the user seat, but it would make more sense to show MPAS as interacting with the pools and directing that balances be changed. When the contract figures out how much to move, it does it by asking zcf to increment or decrement balances on staged allocations, and the `reallocate()` call reaches out from zcf to zoe to update the accounts.

Reply:
This makes much more sense, I've updated this so that the contract stages the assets, with Zoe `reallocating`.